Run helloworld server in build/target wildfly; configure build/target for ssl

ronsigal opened the pull request.

ronsigal commented:
I've changed helloworld/service to run in the WildFly in build/target directory. Also, -Dssl= can be used to configure that WildFly for SSL connections. The possible values are "none", "oneway", and "twoway".

hpehl reviewed (commented):
LGTM @ronsigal! There are some formatting issues, though.

jamezp commented on examples/helloworld/service/pom.xml:
```diff
@@ -3,119 +3,210 @@
+     you may not use   his file except in compliance with   he License.
+     You may obtain a copy of   he License at
 
          https://www.apache.org/licenses/LICENSE-2.0
 
-     Unless required by applicable law or agreed to in writing, software
-     distributed under the License is distributed on an "AS IS" BASIS,
+     Unless required by applicable law or agreed   o in writing, software
+     distributed under   he License is distributed on an "AS IS" BASIS,
      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-     See the License for the specific language governing permissions and
-     limitations under the License.
+     See   he License for   he specific language governing permissions and
```
Looks like all the t's went missing :)

jamezp reviewed (changes requested):
Minor nit, it looks like the POM somehow got re-formatted with 2 spaces instead of 4. While that's not a huge deal, IMO it's best for projects to stick to a standard spacing for their source files.

hpehl commented:
> Minor nit, it looks like the POM somehow got re-formatted with 2 spaces instead of 4. While that's not a huge deal, IMO it's best for projects to stick to a standard spacing for their source files.

Running `./format.sh` should fix that.

jamezp commented:
Is it setup to format POM's?

hpehl commented:
Arrg, it misses formatting POMs right now. Going to see if this can be configured...

hpehl commented:
The [`formatter-maven-plugin`](https://code.revelc.net/formatter-maven-plugin/usage.html) can only format Java source code according to a provided [Eclipse formatter config](https://github.com/wildfly-extras/wildfly-grpc-feature-pack/blob/main/build-config/src/main/resources/etc/eclipse-formatter-config.xml).

It isn't meant to format XML. So the POMs should be reformatted manually to have four space indentions again.

jamezp commented:
> The [`formatter-maven-plugin`](https://code.revelc.net/formatter-maven-plugin/usage.html) can only format Java source code according to a provided [Eclipse formatter config](https://github.com/wildfly-extras/wildfly-grpc-feature-pack/blob/main/build-config/src/main/resources/etc/eclipse-formatter-config.xml).
> 
> It isn't meant to format XML. So the POMs should be reformatted manually to have four space indentions again.

It actually can do it https://code.revelc.net/formatter-maven-plugin/format-mojo.html#configXmlFile. However, it would take a separate execution and configuration to only format POM's. There's also a new feature I contributed named `<includeResources/>` to include resources in being configured.

jamezp commented:
> Cool! Didn't know about that.
> 
> Does (re)formatting XML mean losing the comments, blank lines, etc.? If so, we should avoid that.

It shouldn't no. I'm working on a PR for it now :)

ronsigal commented:
I just learned a couple days ago that Ctrl+Shift+F formats XML in Eclipse. It's not ideal, in the sense that it uses tabs and removes empty lines, but it's easy enough correct. The first time I changed the tabs, I changed "\ t" instead of "\t" and the initial 't's went away. ;-)
